e2e/aks_model.go

```diff
@@ -19,6 +19,37 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/privatedns/armprivatedns"
 )
 
+// getLatestGAKubernetesVersion returns the latest GA Kubernetes version for the given location.
+func getLatestGAKubernetesVersion(location string) (string, error) {
+	versions, err := config.Azure.AKS.ListKubernetesVersions(context.Background(), location, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to list Kubernetes versions: %w", err)
+	}
+	if versions.Values == nil || len(versions.Values) == 0 {
+		return "", fmt.Errorf("no Kubernetes versions available")
+	}
+	for _, k8s_version := range versions.Values {
+		if k8s_version.Version == nil {
+			continue
+		}
+		if k8s_version.IsPreview == nil || !*k8s_version.IsPreview {
```

Comment:
is there also an `IsLTS` flag we need to check for?

Reply:
I might be missing something, though with the current logic it seems like we aren't guaranteed to get the latest? just the first non-preview version returned by the `ListKubernetesVersions` call

Reply:
can we also log out the chosen k8s version for easier debugging?

Reply:
All future versions are LTS now...